docs/features/network-security-controls/admin-network-policy.md

```diff
@@ -1,303 +1,1118 @@
+  BANP in your cluster. Hence use this policy wisely as a wider default
+  guardrail.
+* API semantics for admin network policies are very different from network
+  policies. There is no implicit deny when you create the policy. You have to
+  explicitly define what you need. Because of this it is not possible to
+  completely isolate the pod from everything using admin network policies for
+  ingress traffic. Only peers you can express for a deny for ingress rule are
+  pods; so atmost only "DENY from all other pods" can be expressed as the best
+  case deny for ingress. In case of egress, one can deny to 0.0.0.0/0 and it will
+  act like a default deny to everything.
+* Unlike network policies, admin network policies are cluster-scoped. So use the
+  empty namespace selector with care because it will select ALL namespaces in your
+  cluster including the infrastructure namespaces like kube-system and kapi-server
+  or openshift-*. Best practice to secure your workloads is to use workload
+  specific labels and then use that as the namespace selector instead.
+* `Pass` is a special case action used to delegate policy decision to the
+  network policy layer. If you don't care about interoperability with existing
+  network policies, then you probably don't need to worry about this feature.
+* Since `0-100` is the supported priority range, it is always better when designing
+  your admin policies to use a middle range like `30-70` by leaving some placeholder
+  priorities before and after. Even in the middle range, be careful to leave gaps so
+  that as your infrastructure requirements evolve over time, you are able to insert
+  new ANPs when needed at the right priority level. If you pack your ANPs, then you
+  might need to recreate all of them to accommodate any changes in the future.
+* Each rule can have upto 100 peers. All those peers are added to the same address-set.
+  It is for the best if selectors across multiple rules don't overlap so that different
+  address-sets don't have the same IPs in them. OVN has perf/scale limitations where
+  it doesn't do incremental processing at ovn-controller level in such cases. So any pod
+  level churn causes full processing of OVS flows. See [here](https://issues.redhat.com/browse/OCPBUGS-31385)
+  for more details.
+* `networks` peer in egress rule is meant to select external destinations. It is recommended
+  to use `namespaces` and `pods` peers to select in-cluster peers. Do not use `networks` for
+  podIP level matching.
+* Currently in order to express multi-tenant isolation, you need to do 1 ANP per tenant.
+  There is some future planned work upstream to make this easier for users. See multi tenant
+  isolation section for more details.
+
+## Known Limitations and Design Choices
+
+* Even if API supports upto 1000 as a value for `spec.priority`, **in OVN-Kubernetes
+  we only support upto 99 as a max value**. This is because of known limitation
```

Comment:
Just to be clear, this is non-conformant, and if you don't already have a test for this in the upstream ANP tests, then you should add such a test so ovn-k can fail it.

If you feel that there is no way ovn-k could ever possibly implement this correctly then you should try to get the API changed upstream.

Reply:
yeah I had raised this last year in u/s meeting and decision was to re-evaluate this at a later stage. The maximum OVN can do is upto 200 ANPs not more than that. I will bring this up in next u/s meeting.

Reply:
also I have a test downstream where OVNK fails it. Probably should add one u/s as well -> just thought that given its a CNI problem maybe much better suited for d/s as a limitation

Reply:
> The maximum OVN can do is up to 200 ANPs not more than that.

(The way _the current implementation_ works, it can't do more than 200 ANPs, but there are other possible implementations. AIUI, you currently reserve 10,000 OVN ACL priorities statically (e.g, `priority=23456` is the 56th rule of the ANP with priority 34 or whatever). This is efficient in terms of reprogramming OVN, but wasteful given that very very few ANPs are actually going to have 100 rules, so almost all of the priorities in the reserved range will go unused. A more dynamic mapping of rules to priorities would let you support more policies (in the normal case), though it would require occasionally renumbering existing ACLs to make more space...)

Reply:
> > The maximum OVN can do is up to 200 ANPs not more than that.
> 
> (The way _the current implementation_ works, it can't do more than 200 ANPs, 

OVN can't do more than 200...

> but there are other possible implementations. AIUI, you currently reserve 10,000 OVN ACL priorities statically (e.g, `priority=23456` is the 56th rule of the ANP with priority 34 or whatever). This is efficient in terms of reprogramming OVN, but wasteful given that very very few ANPs are actually going to have 100 rules, so almost all of the priorities in the reserved range will go unused. A more dynamic mapping of rules to priorities would let you support more policies (in the normal case), though it would require occasionally renumbering existing ACLs to make more space...)

yes OVN-Kubernetes can change and tweak it dynamically on the fly, this was also discussed in past and deemed unclear on how useful this is reason being: we end up with another restriction like we support 1000 policies max with max 10 rules in each as worst case scenario? I guess we need data before doing that dynamic computing changes is how it ended IIRC. i.e do ppl prefer LOTS of policies with less rules in each OR do ppl prefer less policies with lots of rules in each; the problem is saying we support 400 or 500 or whatever that number is but we need to support it on original front with 100 rules on each policy even if users might or might not use it... 
I will open a tech debt item to come back to investigate this in the future based on user feedback in case ppl really need 600-700 policies or at least the ranges are needed..

Reply:
I have clarified this point in the docs.